### PR TITLE
Factor out some common test setup functions.

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: lib/osv/tests.py [hunk 1]
```diff
@@ -12,17 +12,46 @@
...
+
+  def __init__(self, path):
+    self.path = path
+    self._repo = pygit2.init_repository(path, True)
+    tree = self._repo.TreeBuilder().write()
+    author = pygit2.Signature('OSV', 'infra@osv.dev')
+    self._repo.create_commit('HEAD', author, author, 'Initial commit', tree, [])
+
+  def add_file(self, path, contents):
+    """Add a file."""
```
Comment (jonathanmetzman) [inline lib/osv/tests.py:42]: Super nit: I think the style guide says to make function docstrings in descriptive style ("Adds a file").

File: lib/osv/tests.py [hunk 2]
```diff
@@ -12,17 +12,46 @@
...
+    oid = self._repo.write(pygit2.GIT_OBJ_BLOB, contents)
+    self._repo.index.add(pygit2.IndexEntry(path, oid, pygit2.GIT_FILEMODE_BLOB))
+    self._repo.index.write()
+
+  def commit(self, author_name, author_email):
+    """Commit."""
```
Comment (jonathanmetzman) [inline lib/osv/tests.py:48]: nit: Maybe "makes a commit."

**Review by jonathanmetzman — APPROVED** [review]

File: lib/osv/tests.py (hunk 1 above)
Comment (oliverchang) [reply lib/osv/tests.py]: Done and fixed every case in this file.

File: lib/osv/tests.py (hunk 2 above)
Comment (oliverchang) [reply lib/osv/tests.py]: Done!

**Pull request merged**